x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/components/header_actions/create_engine_menu_item.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { EuiBetaBadge, EuiContextMenuItem, EuiText, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import { APP_SEARCH_PLUGIN } from '../../../../../../../common/constants';
+import { ENGINE_CREATION_PATH } from '../../../../../app_search/routes';
+import { ESINDEX_QUERY_PARAMETER } from '../../../../../shared/constants';
+import { generateEncodedPath } from '../../../../../shared/encode_path_params';
+import { KibanaLogic } from '../../../../../shared/kibana';
+
+export interface CreateEngineMenuItemProps {
+  indexName?: string;
+  isHiddenIndex?: boolean;
+}
+
+export const CreateEngineMenuItem: React.FC<CreateEngineMenuItemProps> = ({
+  indexName,
+  isHiddenIndex,
+}) => {
+  const engineCreationPath = !indexName
+    ? `${APP_SEARCH_PLUGIN.URL}${ENGINE_CREATION_PATH}`
+    : generateEncodedPath(`${APP_SEARCH_PLUGIN.URL}${ENGINE_CREATION_PATH}?:indexKey=:indexName`, {
+        indexKey: ESINDEX_QUERY_PARAMETER,
+        indexName,
+      });
+
+  return (
+    <EuiFlexGroup alignItems="center" gutterSize="xs">
+      <EuiFlexItem>
+        <EuiContextMenuItem
+          size="s"
```

Comment:
Davey and I found a bug in the `EuiContextMenuPanel` where the size was only applied to the first item. So I added this so that this menu item would match the view and rendering of the menu greatly improved.

Reply:
Nice! Might be worth creating a pull request to [Elastic EUI](https://github.com/elastic/eui) to fix that bug.